doc/tutorials/using_cat_on_up2.rst

```diff
@@ -0,0 +1,157 @@
+
+#. Set CLOS with wrmsr <reg_num> <value>, we want VM1 to use the lower 6 ways of cache,
+   so CLOS0 is set to 0xc0 for the upper 2 ways, and CLOS1 is set to 0x3f for the lower
+   6 ways.
+
+   .. code-block:: none
+
+     ACRN:\>wrmsr  -p1 0xd10  0xc0
+     ACRN:\>wrmsr  -p1 0xd11  0x3f
+
+#. Attach COS1 to PCPU1, because MSR IA32_PQR_ASSOC [bit 63:32], so write 0x100000000
+   to it ti use CLOS1
+
+   .. code-block:: none
+
+     ACRN:\>wrmsr   –p1   0xc8f    0x100000000
+
+Configure CAT for VM with VM Configuration
+##########################################
+
+#. CAT on ACRN can be enabled and configured by modifying source code, so the first
+   step is clone the ACRN source code:
+
+   .. code-block:: none
+
+     $ git clone https://github.com/projectacrn/acrn-hypervisor.git
+     $ cd acrn-hypervisor/hypervisor
+
+#. The predefined cache masks can be found at
+   ``hypervisor/arch/x86/configs/$(CONFIG_BOARD)/board.c``, for UP2 board, that is
+   ``hypervisor/arch/x86/configs/apl-up2/board.c``, you can change the mask values,
+   but note that, CLOS mask must has continuous bits, or #GP can be triggered.
+
+   .. code-block:: none
+     :emphasize-lines: 3,7,11,15
+
+     struct platform_clos_info platform_clos_array[4] = {
+             {
+                     .clos_mask = 0xff,
+                     .msr_index = MSR_IA32_L2_MASK_0,
+             },
+             {
+                     .clos_mask = 0xff,
+                     .msr_index = MSR_IA32_L2_MASK_1,
+             },
+             {
+                     .clos_mask = 0xff,
+                     .msr_index = MSR_IA32_L2_MASK_2,
+             },
+             {
+                     .clos_mask = 0xff,
+                     .msr_index = MSR_IA32_L2_MASK_3,
+             },
+     };
+
+#. Set up CLOS at the VM config. If you want a VM use one of the CLOSs, You need to find its
+   configuration data and modify it. We will take SOS on sharing mode as an example, its
+   configuration data can be found at ``hypervisor/arch/x86/configs/sharing_config.c``
+
+   .. code-block:: none
+     :emphasize-lines: 5,6
+
+     struct acrn_vm_config vm_configs[CONFIG_MAX_VM_NUM] __aligned(PAGE_SIZE) = {
+             {
+                     .type = SOS_VM,
+                     .name = SOS_VM_CONFIG_NAME,
+                     .guest_flags = SOS_VM_CONFIG_GUEST_FLAGS | CLOS_REQUIRED,
+                     .clos = 1,
+                     .memory = {
+                             .start_hpa = 0x0UL,
+                             .size = CONFIG_SOS_RAM_SIZE,
+                     },
+                     .os_config = {
+                             .name = SOS_VM_CONFIG_OS_NAME,
+                     },
+             },
+     };
+
+#. Build the ACRN hypervisor and copy the artifact ``acrn.efi`` to the
+   ``/boot`` directory, how to build please refer to `Build ACRN from Source <https://projectacrn.github.io/latest/getting-started/building-from-source.html#>`
```

Comment:
We shouldn't put `acrn.efi` directly in `/boot`, instead we need to instruct people to 1) make sure the ESP is mounted under `/boot` and 2) put `acrn.efi` under `/boot/EFI/acrn/`

Reply:
When linking to ACRN documentation, you should use :ref: (and use the label at the top of the .rst file) rather than a hard URL:
```suggestion
   ``/boot`` directory, see  :ref:`getting-started-building` but building instructions.
```

Reply:
OK